src/material/form-field/form-field.ts

```diff
@@ -217,6 +218,20 @@ export class MatFormField
   // Unique id for the label element.
   readonly _labelId = `mat-form-field-label-${nextUniqueId++}`;
 
+  /**
+   * Whether the hint element(s) should be persisted when errors are present (hidden with CSS), or should be detached from the DOM altogether (default).
+   */
+  @Input()
+  get persistHints(): boolean {
```

Comment:
I don't think that persisting the hints should be configurable.

Reply:
@crisbeto If persisted hints is the new standard behavior, that will cause existing tests to fail since the harness uses ` _hints = this.locatorForAll('mat-hint, .mat-hint')`, expecting that to return `[]` when in an error state.

For example, https://github.com/angular/components/blob/master/src/material/form-field/testing/shared.spec.ts#L195:

```
  it('should be able to get hint messages of form-field', async () => {
    const formFields = await loader.getAllHarnesses(formFieldHarness);
    expect(await formFields[1].getTextHints()).toEqual(['Hint 1', 'Hint 2']);

    fixture.componentInstance.requiredControl.setValue('');
    dispatchFakeEvent(fixture.nativeElement.querySelector('#with-errors input'), 'blur');
    expect(await formFields[1].getTextHints()).toEqual([]);
  });
```

Should:
1) the test be updated so that `.getTextHints()` is `['Hint 1', 'Hint 2']` in both error and not-error states, or
2) the test stay the same, and the harness instead is updated, e.g. so that `.getTextHints()` filters `_hints` to exclude hidden hints?

Reply:
I think that the harness should be updated to exclude hidden hints.